lib/resolveOverrides.js

```diff
@@ -0,0 +1,90 @@
+var fs = require('fs-extra');
+var path = require('path');
+var aws = require('aws-sdk');
+var autoAuth = require('aws-auto-auth');
+
+
+function Overrides(options){
+  this.options = options;
+  this.overrides = {};
+  if(options.bucket){
+    var credentials = autoAuth(aws, options.profile);
+    this.credentials = credentials;
+    this.S3 = new aws.S3({credentials: credentials});
+  }
+}
+
+
+Overrides.prototype.resolve = function(name){
+  return this.overrides[name];
+}
+
+
+Overrides.prototype.get = function(name){
+  if(this.options.bucket){
+    var request = this.S3.getObject({'Bucket': this.options.bucket, 'Key': this.overrides[name]});
+    return request.createReadStream();
+  }else{
+    return fs.createReadStream(this.overrides[name]);
+  }
+}
+
+
+Overrides.prototype.list = function(callback){
+  var self = this;
```

Comment:
What's the point of `self`?  This function seems to use `self` half the time, and `this` the other half.